lib/index.js

```diff
@@ -20,7 +20,7 @@ module.exports = (app, options) => {
 
   interceptors.forEach((interceptor) => {
     if (interceptor.path) {
-      app.all(interceptor.path, (req, res, next) => {
+      app.use(interceptor.path, (req, res, next) => {
```

Comment:
THE KEY CHANGE, using `middleware` pattern as the recommended way to add interceptors in express.

Reply:
afaik, there are some differences between `app.all` and `app.use`:

### the path parameter

In `app.use`, the `path` means:

> A route will match any path that follows its path immediately with a “/”. For example: app.use('/apple', ...) will match “/apple”, “/apple/images”, “/apple/images/news”, and so on.
> https://expressjs.com/en/4x/api.html#app.use

In `app.all`, the `path` is required, and if we use `app.all('/apple', ...)`, it will only match `/apple` route, we need to specify `/apple/*` to match all routes sharing the same prefix.

### semantic

`app.all` is clearly to say we want to do something for all HTTP verbs targeting the same path, `app.use` is more used in adding middlewares.

I'm not sure whether this change is necessary, and whether the `path` will become quite different with the normal meaning of `app.get`/`app.post` methods. Please have a further consideration and then share your thoughts here.

Reply:
first of all, the interceptors look more likely a `middleware` to me. the reason I changed the `app.all` to `app.use` is that `app.all` will be used as an actual route in `express`, IMO, it probably should be defined at the end of each request, instead of somewhere else. I think each handler that happens in the middle of all handlers needs to use `app.use` instead.

e.g.
```
request `/test/abc` -> app.all() -> app.all() -> app.get()
vs
request `/test/abc` -> app.use() -> app.use() -> app.get()
```

and to answer your questions, I guess maybe I can add an extra option called `exact: true | false` to check whether the request URL is exact matched in the middleware.

e.g.
```
  interceptors.forEach((interceptor) => {
    if (interceptor.path) {
      let matcher = null;
      if (interceptor.exact) {
        matcher = pathToRegexp(interceptor.path);
      }

      app.use(interceptor.path, (req, res, next) => {
        req.interceptorFlags = req.interceptorFlags || [];
        if (!matcher || matcher(req.url)) {
          req.interceptorFlags.push(interceptor.flag);
        }
        next();
      });
    }
  });
```

so that we can use `path-to-regexp` to match the `req.url` in each interceptors middleware.

<img width="921" alt="image" src="https://user-images.githubusercontent.com/5073227/194741819-9d149760-447c-4680-a99b-d2bee4722d93.png">

<img width="895" alt="image" src="https://user-images.githubusercontent.com/5073227/194741687-5b3aaffa-4bf1-4ca8-9de5-7061cc86c141.png">

Reply:
anyway, the reason why I want to change the `app.all` to `app.use` is because `app.all` will add up some extra `route stacks` information in each request, and it'll mess around the tracing information, so we can't easily identify the request path in our tracing apps.

Reply:
Sounds reasonable to me, and probably we could add the `exact` option to be backward compatible.